docs/resources/kafka_native_acl.md

```diff
@@ -3,32 +3,51 @@
+  Kafka-native ACLs provide advanced resource-level access control with fine-grained permissions, including ALLOW and DENY rules. For simplified topic-level control you can use Aiven ACLs https://registry.terraform.io/providers/aiven/aiven/latest/docs/resources/kafka_acl.
 ---
 
 # aiven_kafka_native_acl (Resource)
 
-Manages native acls in [kafka service](https://aiven.io/docs/products/kafka/concepts/acl).
+Creates and manages Kafka-native [access control lists](https://aiven.io/docs/products/kafka/concepts/acl) (ACLs) for an Aiven for Apache Kafka® service. ACLs control access to Kafka topics, consumer groups,
+clusters, and Schema Registry.
 
+Kafka-native ACLs provide advanced resource-level access control with fine-grained permissions, including `ALLOW` and `DENY` rules. For simplified topic-level control you can use [Aiven ACLs](https://registry.terraform.io/providers/aiven/aiven/latest/docs/resources/kafka_acl).
 
+## Example Usage
+
+```terraform
+resource "aiven_kafka_native_acl" "example_acl" {
+  project         = data.aiven_project.example_project.project
+  service_name    = aiven_kafka.example_kafka.service_name
+  resource_type   = "Topic"
+  resource_name   = "example-topic"
+  principal       = "User:example-user"
+  operation      = "Read"
+  pattern_type    = "LITERAL"
+  permission_type = "ALLOW"
+  host            = "198.51.100.0"
+}
+```
 
 <!-- schema generated by tfplugindocs -->
 ## Schema
 
 ### Required
 
-- `operation` (String) The operation. The possible values are `All`, `Alter`, `AlterConfigs`, `ClusterAction`, `Create`, `CreateTokens`, `Delete`, `Describe`, `DescribeConfigs`, `DescribeTokens`, `IdempotentWrite`, `Read` and `Write`. Changing this property forces recreation of the resource.
+- `operation` (String) The action that a principal is allowed or denied on the Kafka resource. The possible values are `All`, `Alter`, `AlterConfigs`, `ClusterAction`, `Create`, `CreateTokens`, `Delete`, `Describe`, `DescribeConfigs`, `DescribeTokens`, `IdempotentWrite`, `Read` and `Write`. Changing this property forces recreation of the resource.
 - `pattern_type` (String) Resource pattern used to match specified resources. The possible values are `LITERAL` and `PREFIXED`. Changing this property forces recreation of the resource.
-- `permission_type` (String) The permission type. The possible values are `ALLOW` and `DENY`. Changing this property forces recreation of the resource.
-- `principal` (String) Principal is in type:name' format. Maximum length: `256`. Changing this property forces recreation of the resource.
+- `permission_type` (String) Specifies whether the action is explicitly allowed or denied for the principal on the specified resource. The possible values are `ALLOW` and `DENY`. Changing this property forces recreation of the resource.
+- `principal` (String) Identities in `type:name` format that the permissions apply to. Maximum length: `256`. Changing this property forces recreation of the resource.
```

Comment:
Only `User:` is supported for the `principal`, and wildcards (for example, `User:alice*`) can be used to match usernames.
cc: @roope-kar

Reply:
Updated.